app/controllers/user_controller.php

```diff
@@ -0,0 +1,85 @@
+<?php 
+class UserController extends AppController
+{
+    public function register() 
+    {
+        if (is_logged_in()) {
+            redirect(url('user/home'));
+        }
+        
+        $params = array(
+            'username' => trim(Param::get('username')),
+            'first_name' => trim(Param::get('first_name')),
+            'last_name' => trim(Param::get('last_name')),
+            'email' => trim(Param::get('email')),
```

Comment:
If you're using the trim() function for the validation, you could just use it on the helper used for validation :bow: 
ex:
```php
function validate_between($check, $min, $max)            
{
    $check = trim($check);
    $n = mb_strlen($check);
                        
    return $min <= $n && $n <= $max;
}
```